api/etcd/get_test.go

```diff
@@ -0,0 +1,69 @@
+package etcd
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/heetch/regula"
+	"github.com/heetch/regula/api"
+	"github.com/heetch/regula/rule"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGet(t *testing.T) {
+	t.Parallel()
+
+	s, cleanup := newEtcdRulesetService(t)
+	defer cleanup()
+
+	path := "p/a/t/h"
+	sig := &regula.Signature{ReturnType: "bool", Params: make(map[string]string)}
+
+	t.Run("OK", func(t *testing.T) {
+		rules1 := []*rule.Rule{rule.New(rule.True(), rule.BoolValue(true))}
+		createBoolRuleset(t, s, path, rules1...)
+
+		ruleset1, err := s.Get(context.Background(), path, "")
+		require.NoError(t, err)
+		require.Equal(t, path, ruleset1.Path)
+		require.Len(t, ruleset1.Versions, 1)
+		require.NotEmpty(t, rules1, ruleset1.Version)
+		require.Equal(t, rules1, ruleset1.Rules)
+		require.Equal(t, sig, ruleset1.Signature)
+
+		// we are waiting 1 second here to avoid creating the new version in the same second as the previous one
+		// ksuid gives a sorting with a one second precision
+		time.Sleep(time.Second)
+		rules2 := []*rule.Rule{rule.New(rule.Eq(rule.StringValue("foo"), rule.StringValue("foo")), rule.BoolValue(true))}
+		createBoolRuleset(t, s, path, rules2...)
+
+		// it should return the second version
+		ruleset2, err := s.Get(context.Background(), path, "")
+		require.NoError(t, err)
+		require.Equal(t, path, ruleset2.Path)
+		require.Len(t, ruleset2.Versions, 2)
+		require.Equal(t, ruleset1.Version, ruleset2.Versions[0])
+		require.Equal(t, ruleset2.Version, ruleset2.Versions[1])
+		require.Equal(t, rules2, ruleset2.Rules)
+		require.Equal(t, sig, ruleset2.Signature)
+
+		// it should return the second version
+		rs, err := s.Get(context.Background(), path, ruleset2.Version)
```

Comment:
I suggest to ask for the 1st version because it is not the one returned by default. Wdyt?